### PR TITLE
Add MOSNS and MOSS algorithms

### PR DESCRIPTION
**Pull request by MadHypnofrog**

### CODE REVIEW

File: ITMO_FS/filters/MOSS.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        array-like : the indices of features to leave
+        """
+
+		newX, newY = SMOTE(X, y)
+		min_rvalue = 1
+		min_b = []
+		for l in np.arange(0.0, 1.0, 0.05):  # TODO: do a little more research on the range of lambdas; random datasets work well with 0-1
+			b = np.array([rnd.random() / 10 for i in range(newX.shape[1])])
+			b0 = rnd.random() / 10
+			for epoch in range(10000):
```
Comment (LastShekel) [inline ITMO_FS/filters/MOSS.py]: see above

File: ITMO_FS/filters/filter_utils.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            The input samples.
+        y : array-like, shape (n_samples)
+            The classes for the samples.
+        k : int
+            The amount of nearest neighbors used in the calculation.
+        theta : int
+            The threshold value: if from k nearest neighbors of an object more than theta of them are of a different class, 
+            then this object is in the overlap region.
+        Returns
+        ------
```
[the remark this answers is not in the document]
Comment (MadHypnofrog) [reply ITMO_FS/filters/filter_utils.py]: done

File: ITMO_FS/filters/MOSNS.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        Returns
+        ------
+        array-like : the indices of features to leave
+        """
+        min_rvalue = 1
+        min_b = []
+        for l in np.arange(0.0, 3.0, 0.1):  # TODO: do a little more research on the range of lambdas
+            b = np.array([rnd.random() / 10 for i in range(X.shape[1])])
+            b0 = rnd.random() / 10
+            for epoch in range(10000):
```
[the remark this answers is not in the document]
Comment (MadHypnofrog) [reply ITMO_FS/filters/MOSNS.py]: done

File: ITMO_FS/filters/MOSS.py (hunk 1 above)
Comment (MadHypnofrog) [reply ITMO_FS/filters/MOSS.py]: done

File: ITMO_FS/filters/MOSNS.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        nu : float
+            The learning rate.
+        threshold : float
+            The threshold value for feature dropout. Instead of comparing them to zero, they are normalized 
+            and values with absolute value lower than the threshold are dropped out.
+        epochs : int
+            The number of epochs to perform in gradient descent.
+        Returns
+        ------
+        array-like : the indices of features to leave
```
Comment (LastShekel) [inline ITMO_FS/filters/MOSNS.py]: Both of these methods should return dataset with selected columns
see Filter/UnivariateFilter

File: ITMO_FS/filters/MOSS.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        nu : float
+        	The learning rate.
+        threshold : float
+        	The threshold value for feature dropout. Instead of comparing them to zero, they are normalized 
+        	and values with absolute value lower than the threshold are dropped out.
+        epochs : int
+            The number of epochs to perform in gradient descent.
+        Returns
+        ------
+        array-like : the indices of features to leave
```
Comment (LastShekel) [inline ITMO_FS/filters/MOSS.py]: Both of these methods should return dataset with selected columns
see Filter/UnivariateFilter

File: ITMO_FS/filters/MOSS.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        """
+
+		newX, newY = SMOTE(X, y)
+		min_rvalue = 1
+		min_b = []
+		for l in np.arange(0.0, 1.0, 0.05):  # TODO: do a little more research on the range of lambdas; random datasets work well with 0-1
+			b = np.array([rnd.random() / 10 for i in range(newX.shape[1])])
+			b0 = rnd.random() / 10
```
Comment (LastShekel) [inline ITMO_FS/filters/MOSS.py]: why we need b0?

File: ITMO_FS/filters/MOSNS.py (hunk 4 above)
Comment (MadHypnofrog) [reply ITMO_FS/filters/MOSNS.py]: fixed

File: ITMO_FS/filters/MOSS.py (hunk 5 above)
Comment (MadHypnofrog) [reply ITMO_FS/filters/MOSS.py]: fixed

File: ITMO_FS/filters/MOSS.py (hunk 6 above)
Comment (MadHypnofrog) [reply ITMO_FS/filters/MOSS.py]: fixed

**Pull request merged**